servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/DefaultRequestTracker.java

```diff
@@ -202,6 +199,11 @@ private void updateEwma(long penalty, long startTimeNanos) {
     }
 
     private static int nanoToMillis(long nanos) {
-        return (int) MILLISECONDS.convert(min(nanos, MAX_MS_TO_NS), NANOSECONDS);
+        return (int) Long.min(Integer.MAX_VALUE, MILLISECONDS.convert(Long.min(nanos, MAX_MS_TO_NS), NANOSECONDS));
+    }
+
+    private static int safeMultiply(int a, int b) {
+        long result = ((long) a) * b;
+        return (int) Long.min(MAX_VALUE, result);
```

Comment:
Line 202 uses `Integer.MAX_VALUE`, so I assumed that this line uses `Long.MAX_VALUE`. Only when I opened it in IDE, I cloud see that this is a static import for `Integer.MAX_VALUE`. If you are specific about Integer/Long min/max, please do the same for imported `MIN_VALUE/MAX_VALUE`. Otherwise, `Long/Integer` changes everything in this method.